Persistent flag / option to turn on logfile logging / Database connector

faisaltheparttimecoder opened the pull request:
# Persistent flag fix / Logfile Redirect

+ The issue earlier seems to be that we were setting the cobra persistent flag via a function, which I guess doesn't work as it doesnt show part of the subcommand, added the same steps as the documentation and this seems to work.

+ since now we have global flags, the LOG LEVEL env variable is now changed to flag verbose.

+ added global flag to save all the log message to the logfile if needed along with destination.

# Database connector

+ A generic wrapper to handle all the database queries, instead of repeating the scan step for each query.
+ This wrapper or module handles database connection and disconnection
+ This module fires the query ( which is supplied ) and it return the data back to the user for further alteration.
+ This makes uses of lib/pq module.

faisaltheparttimecoder committed:
Fix for persistent flag & option to turn on logfile logging

+ The issue earlier seems to be that we were setting the cobra persistent flag via a function, which I guess doesn't work as it doesnt show part of the subcommand, added the same steps as the documentation and this seems to work.

+ since now we have global flags, the LOG LEVEL env variable is now changed to flag verbose.

+ added global flag to save all the log message to the logfile if needed along with destination.

faisaltheparttimecoder committed:
fixing typo

Fixing few typo error's

faisaltheparttimecoder committed:
Database connector wrapper

+ A generic wrapper to handle all the database queries, instead of repeating the scan step for each query.
+ This wrapper or module handles database connection and disconnection
+ This module fires the query ( which is supplied ) and it return the data back to the user for further alteration.
+ This makes uses of lib/pq module.

faisaltheparttimecoder committed:
Merge branch 'master' of https://github.com/pivotal-gss/gpmt2

tylarb commented on cmd/root.go:
```diff
@@ -100,9 +113,24 @@ func flagsLogCollector() {
+	// Database connection parameters.
+	rootCmd.PersistentFlags().StringVarP(&hostname, "hostname", "", "localhost","Hostname where the database is hosted")
+	rootCmd.PersistentFlags().IntVarP(&port, "port", "", 5432, "Port number of the master database")
+	rootCmd.PersistentFlags().StringVarP(&database, "database", "", "template1", "Database name to connect")
```
These can be just `StringVar` instead of `StringVarP` if we don't allow a short option.

faisaltheparttimecoder committed:
Removal of short name command line

Changed the datatype of database global flag connection to the one that doesn't support shorthand options.
